src/pages/login/InnerLoginForm.js

```diff
@@ -0,0 +1,144 @@
+        form.focus()
+        cancelTwoFA()
+        ref?.current?.focus()
+    }
+    const loginButtonText = twoFAVerificationRequired
+        ? i18n.t('Verify and log in', { lngs })
+        : i18n.t('Log in', { lngs })
+    const login2FAButtonText = twoFAVerificationRequired
+        ? i18n.t('Verifying...', { lngs })
+        : i18n.t('Logging in...', { lngs })
+    const isRequired = getIsRequired(lngs[0])
+
+    return (
+        <form onSubmit={handleSubmit}>
+            <div
+                className={
+                    twoFAVerificationRequired ? styles.hiddenFields : null
+                }
+            >
+                {/* onChange will not update every change, so may need to use controlled InputField here for username tracking */}
+                <ReactFinalForm.Field
+                    name="username"
+                    label={i18n.t('Username', { lngs })}
+                    component={InputFieldFF}
+                    className={styles.inputField}
+                    validate={!formSubmitted ? null : isRequired}
+                    key={formSubmitted ? 1 : 0}
+                    initialFocus={!twoFAVerificationRequired}
+                    onBlur={(e) => {
+                        setFormUserName(e.value)
+                    }}
+                    readOnly={loading}
+                />
+                <ReactFinalForm.Field
+                    name="password"
+                    label={i18n.t('Password', { lngs })}
+                    type="password"
+                    component={InputFieldFF}
+                    className={styles.inputField}
+                    validate={!formSubmitted ? null : isRequired}
+                    key={formSubmitted ? 2 : 3}
+                    readOnly={loading}
+                />
+            </div>
+
+            <div
+                className={
+                    !twoFAVerificationRequired ? styles.hiddenFields : ''
+                }
+            >
+                <ReactFinalForm.Field
+                    name="twoFA"
+                    label={i18n.t('Authentication code', { lngs })}
+                    component={InputFieldFF}
+                    className={styles.inputField}
+                    initialValue=""
+                    initialFocus={twoFAVerificationRequired}
+                    readOnly={loading}
+                />
+            </div>
+            <div className={styles.formButtons}>
+                <Button
+                    type="submit"
+                    onClick={verify}
+                    disabled={loading}
+                    primary
+                >
+                    {loading ? login2FAButtonText : loginButtonText}
+                </Button>
+                {showResentCode && (
+                    <Button
+                        secondary
+                        disabled={isResendDisabled || loading}
+                        onClick={resendCode}
+                        loading={loading}
```

Comment:
This will set this button to "loading" spinner when you're logging in, which I don't think is very clear communication to the user (who has no reason to believe that resend code/login are related actions)